src/prototype_creep_fight.js

```diff
@@ -346,75 +346,159 @@ Creep.prototype.siege = function() {
   this.memory.hitsLost = this.memory.hitsLast - this.hits;
   this.memory.hitsLast = this.hits;
 
-  if (this.hits - this.memory.hitsLost < this.hits / 2) {
+  if (this.hits - this.memory.hitsLost < this.hits / 2 && this.room.name == this.memory.routing.targetRoom) {
     let exitNext = this.pos.findClosestByRange(FIND_EXIT);
     this.moveTo(exitNext);
     return true;
   }
-
-  if (!this.memory.notified) {
+  if (config.autoattack.notify && !this.memory.notified) {
     this.log('Attacking');
     Game.notify(Game.time + ' ' + this.room.name + ' Attacking');
     this.memory.notified = true;
   }
-  var tower = this.pos.findClosestByRange(FIND_HOSTILE_STRUCTURES, {
+
+  if (this.room.controller.safeMode) {
+    let constructionSites = this.room.find(FIND_CONSTRUCTION_SITES);
+    this.moveTo(constructionSites[0]);
+    return true;
+  }
+
+  var spawn = this.pos.findClosestByRange(FIND_HOSTILE_STRUCTURES, {
     filter: function(object) {
-      if (object.structureType == STRUCTURE_TOWER) {
-        return true;
-      }
-      if (object.structureType == STRUCTURE_CONTROLLER) {
-        return true;
-      }
-      return true;
+      return object.structureType == STRUCTURE_SPAWN;
     }
   });
-  let target = tower;
-  if (tower === null) {
-    var spawn = this.pos.findClosestByRange(FIND_HOSTILE_STRUCTURES, {
+
+  if (spawn === null) {
+    let structures = this.pos.findClosestByRange(FIND_HOSTILE_STRUCTURES, {
       filter: function(object) {
-        if (object.structureType == 'spawn') {
-          return true;
-        }
-        return false;
+        return object.structureType != STRUCTURE_CONTROLLER;
       }
     });
-    target = spawn;
+
+    if (structures === null) {
+      var constructionSites = this.pos.findClosestByRange(FIND_CONSTRUCTION_SITES);
+      this.moveTo(constructionSites);
+      return true;
+    }
+
+    this.moveTo(structures);
+    this.dismantle(structures);
+    return true;
   }
-  if (target === null) {
-    var cs = this.pos.findClosestByRange(FIND_CONSTRUCTION_SITES);
-    this.moveTo(cs);
-    return false;
+  //  var path = this.pos.findPathTo(spawn, {
+  //    ignoreDestructibleStructures: true
+  //  });
+  let search = PathFinder.search(
+    this.pos, {
+      pos: spawn.pos,
+      range: 1
+    }, {
+      maxRooms: 1
+    }
+  );
+  if (this.pos.getRangeTo(spawn.pos) <= 1) {
+    this.dismantle(spawn);
+  } else if (search.path[0].lookFor(LOOK_STRUCTURES).length !== 0) {
+    let structures = this.pos.findInRange(FIND_STRUCTURES, 1);
+    structures.sort(function(a, b) { return a.hits - b.hits; });
+    this.cancelOrder('dismantle');
+    this.dismantle(structures[0]);
```

Comment:
maybe not, code is a mess, working on the getting it to work part, then I will cleanup :)